### PR TITLE
helper: new temporal helper

### PR DESCRIPTION
**Pull request by mcuadros**
A helper to implement the `billy.TmpFile` interface

### CODE REVIEW

File: test/tempfile.go [hunk 1]
```diff
@@ -20,23 +20,24 @@ func (s *TempFileSuite) TestTempFile(c *C) {
 	f, err := s.FS.TempFile("", "bar")
 	c.Assert(err, IsNil)
 	c.Assert(f.Close(), IsNil)
-	c.Assert(strings.HasPrefix(f.Name(), "bar"), Equals, true)
+
+	c.Assert(strings.Index(f.Name(), "bar"), Not(Equals), -1)
```
Comment (ajnavarro) [inline test/tempfile.go]: Why you change the tests? `bar` is not the prefix now?

Comment (mcuadros) [inline test/tempfile.go]: because, the file not stars always with this path, is not guaranteed.

Comment (ajnavarro) [inline test/tempfile.go]: But, this is a specific case that is testing a specific behavior. Then, If you changed some code, the output of the file system should be the same. 

Another good practice, at least for me, is to have assets the most specific possible. In this case we can swallow a lot of bug related to path names doing this.

File: helper/temporal/temporal.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+// Temporal is a helper that implements billy.TempFile over any filesystem.
+type Temporal struct {
+	billy.Filesystem
+	tmpdir string
+}
+
+// New creates a new filesystem wrapping up 'fs' the intercepts the calls to
+// the TempFile method.
```
Comment (alcortesm) [inline helper/temporal/temporal.go]: The second argument to this function needs to be explained.  Especially how does it modifies the logic of the `TempFile` method first argument.

File: helper/temporal/temporal.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+func New(fs billy.Filesystem, tmpdir string) billy.Filesystem {
+	return &Temporal{
+		Filesystem: fs,
+		tmpdir:     tmpdir,
+	}
+}
+
+func (h *Temporal) TempFile(dir, prefix string) (billy.File, error) {
+	return util.TempFile(h.Filesystem, h.Join(h.tmpdir, dir), prefix)
```
Comment (alcortesm) [inline helper/temporal/temporal.go]: According to the TempFile interface description...
> TempFile creates a new temporary file in the directory dir with a name beginning with prefix...
but I believe this method is creating the file in the directory `h.tmpdir/dir`.

Is this correct?

File: test/tempfile.go (hunk 1 above)
Comment (mcuadros) [inline test/tempfile.go]: a temp file has a random part so can't be asset properly. The paths are covered by hundreds of other tests. 

I prefer having this test in this way that having a other test on the temporal package where this test is shadowed.

File: helper/temporal/temporal.go (hunk 3 above)
Comment (mcuadros) [inline helper/temporal/temporal.go]: true, I just changed the behavior

File: helper/temporal/temporal.go (hunk 2 above)
Comment (mcuadros) [inline helper/temporal/temporal.go]: done

**Pull request merged**